BUILD.openssl.bazel

```diff
@@ -205,27 +238,27 @@ COMMON_OPENSSL_COPTS = [
     "//conditions:default": [],
 }) + select({
```

Comment:
Let's remove this select - if we need it in the future, we can re-add it :)

Reply:
From my own understanding the only thing that changes here is that windows doesn't receive this flag and since we're assuming windows is MSVC `-fPIC` is not an MSVC flag and would result in a complaint.
https://github.com/bazelbuild/bazel-central-registry/issues/5972